### PR TITLE
Fix relative path tool path

### PR DESCRIPTION
**Pull request by wli3**
fix https://github.com/dotnet/cli/issues/9319

Replace PathUtility with new Path.GetRelativePath corefx api

Pass pull path to such api

--------------------

It would like to have a next step later that enforces DirectoryPath and FilePath always store full path. The problem I have is Path behaves differently in windows and mac. For example, /home means different thing in windows and unix. So if I store full path currently, a lot of test will fail. But I think it worth to do the proper fix to eliminate the whole class of bugs

### CODE REVIEW

File: src/Microsoft.DotNet.Cli.Utils/PathUtility.cs [hunk 1]
```diff
@@ -119,6 +119,11 @@ public static string GetRelativePath(DirectoryInfo directory, FileSystemInfo chi
         /// </summary>
         public static string GetRelativePath(string path1, string path2)
         {
+            if (!Path.IsPathRooted(path1) || !Path.IsPathRooted(path2))
```
Comment (wli3) [inline src/Microsoft.DotNet.Cli.Utils/PathUtility.cs]: I still cannot remove this. It is used by testframework and it is netstandard. But at least I can add some guard

File: src/dotnet/ShellShim/AppHostShimMaker.cs [hunk 2]
```diff
@@ -39,8 +39,8 @@ public void CreateApphostShellShim(FilePath entryPoint, FilePath shimPath)
                 appHostSourcePath = Path.Combine(_appHostSourceDirectory, ApphostNameWithoutExtension);
             }
 
-            var appHostDestinationFilePath = shimPath.Value;
-            var appBinaryFilePath = PathUtility.GetRelativePath(appHostDestinationFilePath, entryPoint.Value);
+            var appHostDestinationFilePath = Path.GetFullPath(shimPath.Value);
+            var appBinaryFilePath = Path.GetRelativePath(Path.GetDirectoryName(appHostDestinationFilePath), Path.GetFullPath(entryPoint.Value));
```
Comment (wli3) [inline src/dotnet/ShellShim/AppHostShimMaker.cs]: Path.GetRelativePath need to pass a directory. As suggested by its document

**Comment by peterhuene** [comment]
Do we have a test to cover the execution of the shim so that we can ensure this regression doesn't happen again?

File: src/Microsoft.DotNet.Cli.Utils/PathUtility.cs (hunk 1 above)
Comment (wli3) [reply src/Microsoft.DotNet.Cli.Utils/PathUtility.cs]: Another con I can think of is we cannot dogfood fix netstandard, netcore projects.

**Comment by wli3** [comment]
let me add test

File: test/Microsoft.DotNet.ShellShim.Tests/ShellShimRepositoryTests.cs [hunk 3]
```diff
@@ -45,6 +45,27 @@ public void GivenAnExecutablePathItCanGenerateShimFile()
             stdOut.Should().Contain("Hello World");
         }
 
+        // Reproduce https://github.com/dotnet/cli/issues/9319
```
Comment (wli3) [inline test/Microsoft.DotNet.ShellShim.Tests/ShellShimRepositoryTests.cs]: @peterhuene @nguerrera added test to repro. I reverted 2 commits, see the repro and re apply the 2 commits back, and see it is fixed

**Comment by wli3** [comment]
@dotnet-bot test Ubuntu x64 Release Build

**Comment by wli3** [comment]
🔔  @peterhuene @nguerrera  :)

**Pull request merged**